File: nodejs/src/decorator/SugarDecorator.ts
import BeverageDecorator from "./BeverageDecorator";

class SugarDecorator extends BeverageDecorator {
    getDescription(): string {
        return `${this.beverage.getDescription()} + Đường`;
    }
    getCost(): number {
        return this.beverage.getCost() + 5000;
    }
}

export default SugarDecorator;

File: spring/demo/src/main/java/com/example/demo/decorator/SugarDecorator.java
package com.example.demo.decorator;

public class SugarDecorator extends BeverageDecorator {
    public SugarDecorator(Beverage beverage) {
        super(beverage);
    }

    @Override
    public String getDescription() {
        return beverage.getDescription() + " + Đường";
    }

    @Override
    public int getCost() {
        return beverage.getCost() + 5000;
    }
}